apps/extension/src/entrypoints/options/pages/translation/personalized-prompt/index.tsx

```diff
@@ -43,31 +46,86 @@ export function PersonalizedPrompts() {
 }
 
 function PromptList() {
-  const translateConfig = useAtomValue(configFieldsAtomMap.translate)
+  const [translateConfig, setTranslateConfig] = useAtom(configFieldsAtomMap.translate)
   const promptsConfig = translateConfig.promptsConfig
   const patterns = promptsConfig.patterns
   const [selectedPrompts, setSelectedPrompts] = useState<string[]>([])
+  const [isExportMode, setIsExportMode] = useState(false)
+  const currentPromptId = promptsConfig.prompt
+
+  const setCurrentPromptId = (value: string) => {
+    void setTranslateConfig({
+      promptsConfig: {
+        ...promptsConfig,
+        prompt: value,
+      },
+    })
+  }
 
   return (
     <section className="w-full">
       <header className="w-full text-end mb-4 gap-3 flex justify-end">
-        <ImportPrompts />
-        <ExportPrompts selectedPrompts={selectedPrompts} />
-        <ConfigurePrompt />
+        {
+          isExportMode
+            ? (
+                <>
+                  <Button
+                    variant="outline"
+                    onClick={() => {
+                      setIsExportMode(false)
+                      setSelectedPrompts([])
+                    }}
+                  >
+                    <Icon icon="tabler:x" className="size-4" />
+                    {i18n.t('options.translation.personalizedPrompts.exportPrompt.cancel')}
+                  </Button>
+                  <ExportPrompts
+                    selectedPrompts={selectedPrompts}
+                    setSelectedPrompts={setSelectedPrompts}
+                    setIsExportMode={setIsExportMode}
+                  />
+                </>
+              )
+            : (
+                <>
+                  <ImportPrompts />
+                  <Button
+                    variant="outline"
+                    onClick={() => setIsExportMode(true)}
+                    disabled={patterns.length === 0}
+                  >
+                    <Icon icon="tabler:file-upload" className="size-4" />
+                    {i18n.t('options.translation.personalizedPrompts.export')}
+                  </Button>
+                  <ConfigurePrompt />
+                </>
+              )
+        }
       </header>
-      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-4 max-h-96 overflow-auto p-2">
+      <RadioGroup
```

Comment:
this component is too long, may need to extract it to a single component (can be in this file, but in an extracted component